profiler/src/ProfilerEngine/Datadog.Profiler.Native/SampleValueTypeProvider.cpp

```diff
@@ -0,0 +1,54 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+
+#include "SampleValueTypeProvider.h"
+
+#include <cassert>
+#include <tuple>
+
+
+SampleValueTypeProvider::SampleValueTypeProvider()
+{
+    _sampleTypeDefinitions.reserve(16);
+}
+
+std::vector<SampleValueTypeProvider::Offset> SampleValueTypeProvider::Register(std::vector<SampleValueType> const& valueTypes)
```

Comment:
Rename to GetOrRegister to avoid confusion (at least in my mind ;^)

Reply:
done